### ISSUE TITLE
Fix false positives for empty lines before </style> in max-empty-lines

### ISSUE THREAD

**Issue by ota-meshi** [stylelint/stylelint#3707]
> Clearly describe the bug

There are false positives for empty lines before `</style>` in `max-empty-lines`.

> Which rule, if any, is the bug related to?

`max-empty-lines`

> What CSS is needed to reproduce the bug?

```html
<style>
a {
  color: pink;
}

</style>
```

```html
<style>

</style>
```

> What stylelint configuration is needed to reproduce the bug?

```json
{
  "rules": {
    "max-empty-lines": 1
  }
}
```

> Which version of stylelint are you using?

`9.6.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

CLI with `stylelint **/*.vue`

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

I use standard CSS, but it is in Vue SFC.

> What did you expect to happen?

No warnings to be flagged.

> What actually happened (e.g. what warnings or errors did you get)?

The following warnings were flagged:

```shell
App.vue
XXX:1  ×  Expected no more than 1 empty line   max-empty-lines
```
